sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -1180,4 +1180,26 @@ class SessionCatalog(
     }
   }
 
+  /**
+   * Get an identical copy of the `SessionCatalog`.
+   * The temporary tables and function registry are retained.
+   * The table relation cache will not be populated.
+   */
+  def copy: SessionCatalog = {
+    val catalog = new SessionCatalog(
```

Comment:
if you are copying... shouldnt all the fields be copied?
you should test that any updates to the copied catalog should not get reflected in the earlier catalog.